### PR TITLE
Introduced allow_reset in Inject::Defaults

### PR DESCRIPTION
**Pull request by emaglio**
This is required so we can reset model.class in the Model macro.
Here the related issue: https://github.com/trailblazer/trailblazer-macro/issues/6

### CODE REVIEW

File: lib/trailblazer/operation/inject.rb [hunk 1]
```diff
@@ -19,14 +19,19 @@ def self.Defaults(default_dependencies)
...
+        def initialize(defaults, allow_reset)
           @defaults = defaults
+          @allow_reset = allow_reset
         end
 
         def call((wrap_ctx, original_args), **circuit_options)
           ctx = original_args[0][0]
 
-          @defaults.each { |k, v| ctx[k] ||= v }
+          if @allow_reset
```
Comment (emaglio) [inline lib/trailblazer/operation/inject.rb]: I know that you don't like `if` but it feels necessary here or you suggest a better syntax?

File: lib/trailblazer/operation/trace.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module Trace
       def self.call(operation, *args)
         ctx = PublicCall.options_for_public_call(*args) # redundant with PublicCall::call.
 
-        stack, signal, (ctx, flow_options) = Activity::Trace.(operation, [ctx, {}])
+        stack, signal, (ctx, _flow_options) = Activity::Trace.(operation, [ctx, {}])
```
Comment (emaglio) [inline lib/trailblazer/operation/trace.rb]: just rubocop stuff

**Comment by emaglio** [comment]
Nick introduced a better solution after activity linear refactoring

**Comment by apotonick** [comment]
Here's an example of how to use a defaulting injection now: https://github.com/trailblazer/trailblazer-macro/blob/f3e9ffaa4bbc8a8044bff69f6c5029634d567860/lib/trailblazer/macro/model.rb#L5

The scope of the injection is only around the task that needs injection, after that, the injection is "removed".

**Pull request closed without merging**